### PR TITLE
Chapter recurrent neural networks/language models and dataset

### PR DESCRIPTION
**Pull request by zhuyuanxiang**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-883/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

**Comment by mli** [comment]
Job d2l-zh/PR-883/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

**Comment by mli** [comment]
Job d2l-zh/PR-883/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

**Comment by mli** [comment]
Job d2l-zh/PR-883/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

**Comment by mli** [comment]
Job d2l-zh/PR-883/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

File: chapter_recurrent-neural-networks/language-models-and-dataset.md [hunk 1]
```diff
@@ -1,33 +1,33 @@
 # 语言模型和数据集
 :label:`sec_language_model`
 
-在 :numref:`sec_text_preprocessing` 中，我们了解了如何将文本数据映射到标记中，其中这些标记可以被视为一系列离散的观测，例如单词或字符。假设长度为 $T$ 的文本序列中的标记依次为 $x_1, x_2, \ldots, x_T$。然后，在文本序列中，$x_t$($1 \leq t \leq T$) 可以被认为是时间步 $t$ 处的观测或标签。给定这样的文本序列，*语言模型*（language model）的目标是估计序列的联合概率
+在 :numref:`sec_text_preprocessing` 中，我们了解了如何将文本数据映射为标记，以及这些标记可以被视为一系列离散的观测，例如单词或字符。假设长度为 $T$ 的文本序列中的标记依次为 $x_1, x_2, \ldots, x_T$。于是，$x_t$($1 \leq t \leq T$) 可以被认为是文本序列在时间步 $t$ 处的观测或标签。在给定这样的文本序列时，*语言模型*（language model）的目标是估计序列的联合概率
 
 $$P(x_1, x_2, \ldots, x_T).$$
 
-语言模型是非常有用的。例如，只需要一次抽取一个标记 $x_t \sim P(x_t \mid x_{t-1}, \ldots, x_1)$，一个理想的语言模型就能够基于自己生成自然文本。与猴子使用打字机完全不像的是，从这样的模型中出现的所有文本都将作为自然语言来传递，例如英语文本。此外，只需要基于前面的对话片断中的文本，就足以生成一个有意义的对话。显然，离设计这样的系统还很遥远，因为它需要“理解”文本，而不仅仅是生成在语法上合理的内容。
+语言模型是非常有用的。例如，只需要一次抽取一个标记 $x_t \sim P(x_t \mid x_{t-1}, \ldots, x_1)$，一个理想的语言模型就能够基于模型本身生成自然文本。与猴子使用打字机完全不同的是，从这样的模型中浮现的所有文本都将作为自然语言（例如，英语文本）来传递。此外，只需要基于前面的对话片断中的文本，就足以生成一个有意义的对话。显然，我们离设计出这样的系统还很遥远，因为它需要 *理解* 文本，而不仅仅是生成在语法上合理的内容。
```
Comment (goldmermaid) [inline chapter_recurrent-neural-networks/language-models-and-dataset.md:8]: Hey can you recover back to “理解”?  From the feedback from the publisher of our book, they suggest to only use `*...*` when introduce a new concept.

File: chapter_recurrent-neural-networks/language-models-and-dataset.md [hunk 2]
```diff
@@ -53,11 +53,11 @@ P(x_1, x_2, x_3, x_4) &=  P(x_1) P(x_2  \mid  x_1) P(x_3  \mid  x_1, x_2) P(x_4
 \end{aligned}
 $$
 
-涉及一个、两个和三个变量的概率公式通常分别称为“一元语法”（unigram）、“二元语法”（bigram）和“三元语法”（trigram）。下面，我们将学习如何设计更好的模型。
+通常，涉及一个、两个和三个变量的概率公式分别被称为“一元语法”（unigram）、“二元语法”（bigram）和“三元语法”（trigram）模型。下面，我们将学习如何去设计更好的模型。
 
 ## 自然语言统计
 
-让我们看看如何对真实数据起作用。根据 :numref:`sec_text_preprocessing` 中介绍的时光机器数据集构建词汇表，并打印最常用的10个单词。
+让我们看看在真实数据上效果如何。根据 :numref:`sec_text_preprocessing` 中介绍的时光机器数据集构建词汇表，并打印前 $10$ 个最常用的（频率最高的）单词。
```
Comment (goldmermaid) [inline chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: Hey, please don't worry about adding $...$ to the text. The publisher don't want us include any extra space in the text... Please [PR 868](https://github.com/d2l-ai/d2l-zh/pull/868) for more detail.

Comment (goldmermaid) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: (I know the space problem may exist everywhere...) We will remove the space for all the sections together.

File: chapter_recurrent-neural-networks/language-models-and-dataset.md [hunk 3]
```diff
@@ -137,53 +137,53 @@ d2l.plot([freqs, bigram_freqs, trigram_freqs], xlabel='token: x',
...
 def seq_data_iter_random(corpus, batch_size, num_steps):  #@save
     """使用随机抽样生成一个小批量子序列。"""
-    # 从随机偏移量（包括`num_steps - 1`）开始对序列进行分区
+    # 从随机偏移量开始对序列进行分区，随机范围包括 `num_steps - 1`
     corpus = corpus[random.randint(0, num_steps - 1):]
-    # 减去1，因为我们需要考虑标签
+    # 减去1，是因为我们需要考虑标签
     num_subseqs = (len(corpus) - 1) // num_steps
-    # 长度为`num_steps`的子序列的起始索引
+    # 长度为 `num_steps` 的子序列的起始索引
```
Comment (goldmermaid) [inline chapter_recurrent-neural-networks/language-models-and-dataset.md:167]: Hey @zhuyuanxiang , can you help remove the space in the code comment? (I know it's weird, but the publisher don't want the spaces.) Thanks!

**Comment by mli** [comment]
Job d2l-zh/PR-883/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 2 above)
Comment (zhuyuanxiang) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: If I should remove these space between Chinese characters and English words/special fonts?
I added these spaces in Typora editor for good vision which can help me to find errors.

**Comment by mli** [comment]
Job d2l-zh/PR-883/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 2 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: I see. You don't need to worry too much on that, I will figure out a way to remove the spaces for all section later. Thanks!

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 3 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:167]: While, please help remove any space in the code comments! Thanks!

Comment (zhuyuanxiang) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md]: I have done it.

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 2 above)
Comment (zhuyuanxiang) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: One more question, could you give me a principle about when and where I can use space between these marks(e.g. reference, mathematics equation, code in the main body, etc.)?
Now, I do it based on my visual effect, but I thought it is not a normative way.
Thks.

**Comment by mli** [comment]
Job d2l-zh/PR-883/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-883/

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 3 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md]: Thanks!

File: chapter_recurrent-neural-networks/language-models-and-dataset.md (hunk 2 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/language-models-and-dataset.md:60]: Great question! Just make some notes in style_guide (PR #889). Please let me know if that is clear enough to you?!

**Pull request merged**